### ISSUE TITLE
Encoding issue on binary files

### ISSUE THREAD

**Issue by cmealares** [pashky/restclient.el#140]
Hello,

I call a service that generates a pdf file. The file I receive is corrupted and I believe this is due to an encoding problem.

The problem does not come from the service: I have checked with postman and it works (that's another restclient which I want to ditch in favor of emacs). 
It does not come from browse-url with which I am able to open a pdf file.

I have seen in "Known issues" that you suspect this problem already.

Thank you for this fantastic tool.

**Comment by cmealares** [comment]
Another test I have just done: I have called restclient-copy-curl-command to generate a curl command to get the pdf file. 

The generated command uses quote (') as string delimiter. I am on windows and this does not work. So I have replaced the quotes with double-quotes ("). 

Then I run the command and redirect the output to foo.pdf. It works perfectly.
I believe using double-quotes is more portable.

**Comment by pashky** [comment]
This tool isn't really good choice if you only need to get binary files. It tries to preprocess output buffer, adds response headers etc. It does handle some images, but that's about it. If you need your binary intact, you're much better off using curl.

As for quotes, double quotes will require special characters escaping (e.g. $ are not uncommon in HTTP names). You're welcome to submit a PR doing it correctly.

**Comment by cmealares** [comment]
I do not only need to get binary files. I am sure that this tool can be a complete replacement of postman. It almost works for binary files.

I thought that restclient-http-send-current-raw would be the right function to call but it does some kind of decoding. I managed to make it work by editing restclient-decode-response. I looked at what it does for images and just did the same for pdf. I've added a pdf? variable :
```
         (image? (save-excursion
                   (search-forward-regexp "Content-Type.*[Ii]mage" nil t)))
	 (pdf? (save-excursion
                   (search-forward-regexp "Content-Type.*pdf" nil t)))
         (encoding (if (save-excursion
                         (search-forward-regexp charset-regexp nil t))
                       (intern (downcase (match-string 1)))
                     'utf-8)))
    (if (or image? pdf?)
```

Then, I run the command. I change the eol style of the result buffer to unix (because I am on windows and it would add CR LF otherwise). I save it and it works. 

That is a quick hack but it shows that it can work. Maybe restclient could let the user specify an output file just as it does for file upload (?)  

Regarding the curl quotes/double quotes issue, I will think about it but since I know very little about http I have to do some reading first (no idea about what to escape).
